misk/src/main/kotlin/misk/MiskServiceModule.kt

```diff
@@ -30,6 +30,18 @@ class MiskServiceModule : KAbstractModule() {
   @Provides
   @Singleton
   fun provideServiceManager(services: List<Service>): ServiceManager {
+    // Confirm all services have been registered as singleton. If they aren't singletons,
+    // _readiness checks will fail
+    val invalidServices = services.map { it.javaClass }.filter { !isSingleton(it) }.map { it.name }
+    check(invalidServices.isEmpty()) {
+      "the following services are not marked as @Singleton: ${invalidServices.sorted().joinToString(
+          ", ")}"
+    }
+
     return CoordinatedService.coordinate(services)
   }
+
+  private fun isSingleton(clazz: Class<*>) =
```

Comment:
I wonder what would happen if you injected the `List<Service>` twice and compared they are the same instances to check for singleton scoping

Reply:
Yeah debated that approach. Will consider